### PR TITLE
Implement ZSTD (.zst) support

### PR DESCRIPTION
**Pull request by vitlav**

### CODE REVIEW

File: patoolib/util.py [hunk 1]
```diff
@@ -380,6 +384,7 @@ def get_file_mime_encoding (parts):
     "FLAC audio bitstream data": "audio/flac",
     "MS Windows HtmlHelp Data": "application/x-chm",
     "ZPAQ stream": "application/zpaq",
+    "ZStandart compressed data": "application/zstd",
```
Comment (The-Compiler) [inline patoolib/util.py:387]:
```suggestion
    "ZStandard compressed data": "application/zstd",
```

**Review by The-Compiler — COMMENTED** [review]
Just found this while looking for a more modern alternative to [atool](https://www.nongnu.org/atool/) supporting zstd, since my distribution (Archlinux) compresses packages that way.

Would love to see this merged! Let me know if I there's anything I can do to help getting this in.

**Comment by jruzicka-nic** [comment]
I'd love to see this for exactly the same reasons as @The-Compiler (`atool` got last release in 2012) so I gave this change a test run.

I hit an issue with `patool extract` on Arch package which ends with `.pkg.tar.zst` - it results in `.tar` file being extracted as opposed to extracting its contents as with other `.tar.*` archives.

**Comment by jruzicka-nic** [comment]
I'm getting FAILs for several tar-related tests and these two that seem to be related to new code:

```
FAILED tests/archives/test_zstd.py::TestZStd::test_zstd - patoolib.util.PatoolError: file `patool/tests/data/t.txt.zst' was not found
FAILED tests/archives/test_zstd.py::TestZStd::test_zstd_file - patoolib.util.PatoolError: file `patool/tests/data/t.txt.zst.foo' was not found
```

Is `py.test` green on your local setup, @vitlav ? Also, is `patool extract foo.tar.gz` working for you as expected?

**Comment by wummel** [comment]
this is fixed in commit 3156199475f4673eaf3d9e233cb3d37e1d913f6f

**Pull request closed without merging**